x/stake/keeper.go

```diff
@@ -111,22 +116,10 @@ func (k Keeper) setCandidate(ctx sdk.Context, candidate Candidate) {
 			return
 		}
 
-		// if this candidate wasn't just bonded then update the height and counter
-		if oldCandidate.Status != Bonded {
-			candidate.ValidatorBondHeight = ctx.BlockHeight()
-			counter := k.getIntraTxCounter(ctx)
-			candidate.ValidatorBondCounter = counter
-			k.setIntraTxCounter(ctx, counter+1)
-		}
-
 		// delete the old record in the power ordered list
 		store.Delete(GetValidatorKey(oldCandidate.validator()))
 	}
 
-	// set the new candidate record
-	bz = k.cdc.MustMarshalBinary(candidate)
```

Comment:
Why are we not updating the candidate here?! this it the most crucial store update of set candidate

Reply:
Already updated ten lines above, we don't need to update twice.